server/modules/nodeDef/repository/nodeDefRepository.js

```diff
@@ -31,23 +31,13 @@ const nodeDefSelectFields =
 
 export const insertNodeDef = async (surveyId, nodeDef, client = db) => {
   const parentUuid = NodeDef.getParentUuid(nodeDef)
-  const parentH = parentUuid ?
-    await client.one(
-      `SELECT meta->'h' as h FROM ${getSurveyDBSchema(surveyId)}.node_def WHERE uuid = $1`,
-      [parentUuid]
-    ) : {}
-
-  const meta = {
-    h: R.isEmpty(parentH) ? [] : R.append(parentUuid, parentH.h)
-  }
-
   return await client.one(`
         INSERT INTO ${getSurveyDBSchema(surveyId)}.node_def 
           (parent_uuid, uuid, type, props_draft, meta)
         VALUES ($1, $2, $3, $4, $5::jsonb)
         RETURNING *
     `,
-    [parentUuid, NodeDef.getUuid(nodeDef), NodeDef.getType(nodeDef), NodeDef.getProps(nodeDef), meta],
+    [parentUuid, NodeDef.getUuid(nodeDef), NodeDef.getType(nodeDef), NodeDef.getProps(nodeDef), JSON.stringify(NodeDef.getMeta(nodeDef))],
```

Comment:
Maybe the DB driver should do the conversion to JSON/JSONB?

Reply:
Can you also inline parentUuid and avoid to write return...?

Reply:
Ok